model/client4.go

```diff
@@ -3407,7 +3492,14 @@ func (c *Client4) GetFlaggedPostsForUserInChannel(userId string, channelId strin
 		return nil, BuildErrorResponse(r, err)
 	}
 	defer closeBody(r)
-	return PostListFromJson(r.Body), BuildResponse(r)
+	var list PostList
+	if r.StatusCode == http.StatusNotModified {
```

Comment:
This checking of the response code seems like a behavior change. Is that okay?

Reply:
Before, the umarshal would silently fail (error was not handled) and an empty struct returned. I actually had to add that check to fix some tests from failing.